### PR TITLE
Refactor Viewer Javascript

### PR DESCRIPTION
**Pull request by tpendragon**
This should make it easier to look at and add the timer code to.

### CODE REVIEW

File: app/javascript/viewer/uv_manager.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    this.bindLogin()
+    this.bindResize()
+    this.uvElement.hide()
+    this.loadUV()
+  }
+
+  loadUV () {
+    this.checkManifest().done(this.createUV.bind(this)).fail(this.requestAuth.bind(this))
+  }
+
```
Comment (christinach) [inline app/javascript/viewer/uv_manager.js:13]: 👍

File: app/javascript/viewer/uv_manager.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      configUri: this.configURI,
+      collectionIndex: Number(this.urlDataProvider.get('c', 0)),
+      manifestIndex: Number(this.urlDataProvider.get('m', 0)),
+      sequenceIndex: Number(this.urlDataProvider.get('s', 0)),
+      canvasIndex: Number(this.urlDataProvider.get('cv', 0)),
+      rangeId: this.urlDataProvider.get('rid', 0),
+      rotation: Number(this.urlDataProvider.get('r', 0)),
+      xywh: this.urlDataProvider.get('xywh', ''),
+      embedded: true
+    }, this.urlDataProvider)
```
Comment (christinach) [inline app/javascript/viewer/uv_manager.js:33]: 👍

File: app/javascript/viewer/uv_manager.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      var titleMatch = /title="(.+?)"/.exec(linkHeader)
+      if (titleMatch[1]) {
+        var title = titleMatch[1]
+        var titleElement = document.getElementById('title')
+        titleElement.textContent = title
+        titleElement.style.display = 'block'
+        this.resize()
+      }
+    }
+  }
```
Comment (christinach) [inline app/javascript/viewer/uv_manager.js:61]: 👍

File: app/javascript/viewer/uv_manager.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    const windowHeight = window.innerHeight
+    const titleHeight = $('#title').outerHeight($('#title').is(':visible'))
+    this.uvElement.width(windowWidth)
+    this.uvElement.height(windowHeight - titleHeight)
+  }
+
+  bindResize () {
+    $(window).on('resize', () => this.resize())
+    this.resize()
+  }
```
Comment (christinach) [inline app/javascript/viewer/uv_manager.js:74]: 👍

File: app/javascript/viewer/uv_manager.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    return new UV.URLDataProvider(false)
+  }
+
+  get manifest () {
+    return this.urlDataProvider.get('manifest')
+  }
+
+  get uvElement () {
+    return $('#uv')
+  }
```
Comment (christinach) [inline app/javascript/viewer/uv_manager.js:101]: 👍

File: app/javascript/viewer/uv_manager.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    })
+  }
+
+  get urlDataProvider () {
+    return new UV.URLDataProvider(false)
+  }
+
+  get manifest () {
+    return this.urlDataProvider.get('manifest')
+  }
```
Comment (christinach) [inline app/javascript/viewer/uv_manager.js:97]: 👍

File: app/views/layouts/viewer_layout.html.erb [hunk 7]
```diff
@@ -10,7 +10,7 @@
   </head>
   <body>
     <%= yield %>
-    <%= javascript_include_tag 'viewer' %>
+    <%= javascript_pack_tag 'viewer' %>
     <script type="text/javascript" src="/uv/uv.js"></script>
```
Comment (christinach) [inline app/views/layouts/viewer_layout.html.erb:14]: 👍

**Comment by christinach** [comment]
@tpendragon I'm testing figgy-staging https://figgy-staging.princeton.edu/catalog/85a21f6e-46b7-4fbf-977f-61351e0cba7c and https://catalog-staging.princeton.edu/catalog/8555444 (I have changed in both ol staging boxex to look figgy-staging). The viewer on catalog-staging raises a 404 `Failed to load resource: the server responded with a status of 404 ()`

**Comment by christinach** [comment]
I don't even think that this catalog record (https://catalog-staging.princeton.edu/catalog/8555444) should have a viewer. https://catalog.princeton.edu/catalog/8555444

**Comment by tpendragon** [comment]
The record you're testing with needed to be marked complete.

**Comment by christinach** [comment]
I just updated the graphql to look figgy-staging.

**Comment by christinach** [comment]
@tpendragon thank you for marking it complete. I can see the viewer now.

**Comment by tpendragon** [comment]
> It would be so nice to write some tests, maybe as part of the next step with this javascript?

It's definitely doable now. I'll make a ticket, but it'll probably be in the backlog.

**Pull request merged**